src/Monolog/Handler/Curl/Util.php

```diff
@@ -33,12 +33,18 @@ public static function execute($ch, $retries = 5, $closeAfterDone = true)
     {
         while ($retries--) {
             if (curl_exec($ch) === false) {
-                if (false === in_array(curl_errno($ch), self::$retriableErrorCodes, true) || !$retries) {
+                
+                $curlErrno = curl_errno($ch);
+                
+                if (false === in_array($curlErrno, self::$retriableErrorCodes, true) || !$retries) {
+                    
+                    $curlError = curl_error($ch);
+                    
                     if ($closeAfterDone) {
                         curl_close($ch);
```

Comment:
wouldn't it make more sense to make sure that after you close `$ch` no more `curl_error` / `curl_errno`  is called?